lua/modes.lua

```diff
@@ -198,19 +195,31 @@ M.setup = function(opts)
 				end
 			end
 
+			if key == utils.replace_termcodes('<c-w>') then
+				operator_started = true
+			end
+
 			if
-				(key:lower() == 'v' or key == utils.replace_termcodes('<c-v>'))
-				and not operator_started
+				key:lower() == 'v'
+				or key == utils.replace_termcodes('<c-v>')
 			then
-				M.highlight('visual')
+				if operator_started then
+					M.reset()
+				else
+					M.highlight('visual')
+					operator_started = true
+				end
 			end
 		end
 
 		if
 			current_mode:lower() == 'v'
 			or current_mode == utils.replace_termcodes('<c-v>')
 		then
-			if key == utils.replace_termcodes('<esc>') then
+			if
+				key == utils.replace_termcodes('<esc>')
+				or key == current_mode
```

Comment:
Does this need to be `:lower()`? Might not matter

```suggestion
				or key == current_mode:lower()
```